### PR TITLE
[DOCS] GCP Deployment Pattern

### PR DESCRIPTION
**Pull request by Shinnnyshinshin**
Changes proposed in this pull request:
- Deployment Pattern Document `How to Use Great Expectations with Google Cloud Platform and BigQuery`
- Example `great_expectations/` folder found in `/tests/integration/fixtures/gcp_deployment/great_expectations/` that contains the following: 
  - `great_expectations.yml` that contains configuration for `ExpectationsStore`, `ValidationsStore` and `DataDocs` on Google Cloud Storage (GCS).
  - 2 Checkpoints that will run GE using GCS as the Metadata Store. One will use a BigQuery datasource (`bigquery_checkpoint`) and one will use a GCS datasource (`gcs_checkpoint`)
  - 2 Airflow DAGs that will run the above-mentioned checkpoint in a `BashOperator`. 
  - This folder (after replacing the placeholder values like `<YOUR_GCS_BUCKET_HERE>`) can be directly migrated over to Google Cloud Composer and run using the instructions described in the Deployment Pattern Document. 
- 2 Integration tests that will generate (at runtime) and test the to the above-mentioned  `great_expectations/` configuration and is referred to in the document.  

Previous Design Review notes:
- GREAT-401
- **Reviewers please view JIRA task GREAT-401 for credentials and paths to buckets and databases that can be used for the Friction Logging**

### Definition of Done
- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/en/latest/contributing/testing.html#contributing-testing-writing-unit-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

File: docs/deployment_patterns/how_to_use_great_expectations_in_databricks.md [hunk 1]
```diff
@@ -20,8 +20,8 @@ There are several ways to set up Databricks, this guide centers around an AWS de
 
 We will cover a simple configuration to get you up and running quickly, and link to our other guides for more customized configurations. For example:
   - If you want to validate files stored in DBFS select one of the "File" tabs below. You can also [watch our video walkthrough](https://youtu.be/9mIsmyuJzhQ) of these steps.
-    - If you are using a different file store (e.g. s3, GCS, ABS) take a look at our how-to guides in the "Cloud" section of "Connecting to Your Data" for example configurations. 
-  - If you already have a spark dataframe loaded, select one of the "Dataframe" tabs below.
```
Comment (Shinnnyshinshin) [inline docs/deployment_patterns/how_to_use_great_expectations_in_databricks.md:24]: This file had some minor formatting changes that were introduced as part of `lint`.

File: docs/guides/setup/configuring_data_docs/how_to_host_and_share_data_docs_on_gcs.md [hunk 2]
```diff
@@ -33,29 +33,29 @@ This guide will explain how to host and share Data Docs on Google Cloud Storage.
 
   **app.yaml:**
 
-    ```yaml file=../../../../tests/integration/docusaurus/setup/configuring_data_docs/how_to_host_and_share_data_docs_on_gcs.py#L52-L54
+    ```yaml file=../../../../tests/integration/docusaurus/setup/configuring_data_docs/how_to_host_and_share_data_docs_on_gcs.py#L54-L56
```
Comment (Shinnnyshinshin) [inline docs/guides/setup/configuring_data_docs/how_to_host_and_share_data_docs_on_gcs.md:36]: This file also had some minor line-number changes introduced as part of `lint`

File: sidebars.js [hunk 3]
```diff
@@ -288,6 +288,7 @@ module.exports = {
       items: [
         'deployment_patterns/how_to_instantiate_a_data_context_hosted_environments',
         'deployment_patterns/how_to_use_great_expectations_in_databricks',
+        'deployment_patterns/how_to_use_great_expectations_with_google_cloud_platform_and_bigquery',
```
Comment (Shinnnyshinshin) [inline sidebars.js:291]: Please let me know if this should be introduced as part of a separate PR

**Review by NathanFarmer — CHANGES_REQUESTED** [review]
I left a few comments in my friction log that will need to be addressed and I am currently performing part 2 of the friction logging.

File: docs/deployment_patterns/how_to_use_great_expectations_with_google_cloud_platform_and_bigquery.md [hunk 4]
```diff
@@ -0,0 +1,466 @@
...
+pip install great-expectations --upgrade
+```
+
+### 2. Connect to Metadata Stores on GCP
+
+The following sections describe how you can take a basic local configuration of Great Expectations and connect it to Metadata stores on GCP.
+
+The full configuration used in this guide can be found in the [`great-expectations` repository](https://github.com/great-expectations/great_expectations/tests/integration/fixtures/gcp_deployment/) and is also linked at the bottom of this document.
+
+:::note Note on Trailing Slashes in Metadata Store prefixes
```
Comment (NathanFarmer) [inline docs/deployment_patterns/how_to_use_great_expectations_with_google_cloud_platform_and_bigquery.md:70]: ❤️

File: docs/deployment_patterns/how_to_use_great_expectations_with_google_cloud_platform_and_bigquery.md [hunk 5]
```diff
@@ -0,0 +1,466 @@
...
+
+We will now take the local GE configuration from [Part 1](#part-1-local-configuration-of-great-expectations-that-connects-to-google-cloud-platform) and migrate it to a Cloud Composer environment so that we can automate the workflow.
+
+There are a number of ways that Great Expectations can be run in Cloud Composer or Airflow.
+
+1. [Running a Checkpoint in Airflow using a `bash operator`](https://docs.greatexpectations.io/docs/deployment_patterns/how_to_run_a_checkpoint_in_airflow.html#option-1-running-a-checkpoint-with-a-bashoperator)
+2. [Running a Checkpoint in Airflow using a `python operator`](https://docs.greatexpectations.io/docs/deployment_patterns/how_to_run_a_checkpoint_in_airflow.html#option-2-running-the-checkpoint-script-output-with-a-pythonoperator)
+3. [Running a Checkpoint in Airflow using a `Airflow operator`](https://github.com/great-expectations/airflow-provider-great-expectations)
+
+For our example, we are going to use the `bash operator` to run the Checkpoint. This portion of the guide can also be found in the following [Walkthrough Video](https://drive.google.com/file/d/1YhEMqSRkp5JDIQA_7fleiKTTlEmYx2K8/view?usp=sharing).
```
Comment (NathanFarmer) [inline docs/deployment_patterns/how_to_use_great_expectations_with_google_cloud_platform_and_bigquery.md:332]: This video is very helpful thank you 🙇🏻

File: docs/deployment_patterns/how_to_use_great_expectations_with_google_cloud_platform_and_bigquery.md [hunk 6]
```diff
@@ -0,0 +1,466 @@
...
+:::info Note on Versions.
+The current Deployment Guide was developed and tested in Great Expectations 0.13.49, Composer 1.17.7 and Airflow 2.0.2. Please ensure your Environment is equivalent or newer than this configuration.
+:::
+
+### 3. Install Great Expectations in Cloud Composer
+
+Installing Python dependencies in Cloud Composer can be done through the Composer web Console (recommended), `gcloud` or through a REST query.  Please follow the steps described in [Installing Python dependencies in Google Cloud](https://cloud.google.com/composer/docs/how-to/using/installing-python-dependencies#install-package) to install `great-expectations` in Cloud Composer. If you are connecting to data in BigQuery, please ensure `pybigquery` is also installed in your Cloud Composer environment.
+
+:::info Troubleshooting Installation
+If you run into trouble while installing Great Expectations in Cloud Composer, the [official Google Cloud documentation offers the following guide on troubleshooting PyPI package installations.](https://cloud.google.com/composer/docs/troubleshooting-package-installation)
```
Comment (NathanFarmer) [inline docs/deployment_patterns/how_to_use_great_expectations_with_google_cloud_platform_and_bigquery.md:365]: Thank you for this link

**Review by NathanFarmer — APPROVED** [review]
This is one of the most detailed docs we have! Great work!

**Review by donaldheppner — APPROVED** [review]

**Pull request merged**